android/cli/commands/_build.js

```diff
@@ -2510,6 +2510,113 @@ AndroidBuilder.prototype.copyResources = function copyResources(next) {
 		}
 	}, this);
 
+	// genereate missing app icons
+	tasks.push(function (cb) {
+		this.currentBuildManifestFiles = this.currentBuildManifestFiles || {};
+
+		var lookup = {
+				// ../Resources/android/appicon.png
+				'default':	{ height: 128, width: 128, required: true },
+				// ../platform/android/res/drawable-[key]/appicon.png
+				'ldpi':		{ height: 36, width: 36, required: false },
+				'mdpi':		{ height: 48, width: 48, required: false },
+				'hdpi':		{ height: 72, width: 72, required: false },
+				'xhdpi':	{ height: 96, width: 96, required: false },
+				'xxhdpi':	{ height: 144, width: 144, required: false },
+				'xxxhdpi':	{ height: 192, width: 192, required: false }
+			},
+			possibleDefaultIcons = [
+				path.join(this.projectDir, 'platform', 'android', 'DefaultIcon.png'),
+				path.join(this.projectDir, 'DefaultIcon-android.png'),
+				path.join(this.projectDir, 'DefaultIcon.png')
+			],
+			missingIcons = [],
+			defaultIcon,
+			previousBuildManifestFiles = this.buildManifest && this.buildManifest.files,
+			defaultIconChanged = false;
+
+		possibleDefaultIcons.some(function (icon) {
+			if (fs.existsSync(icon)) {
+				defaultIcon = icon;
+				return true;
+			}
+		});
+
+		if (defaultIcon) {
+			var currDefaultIconHash = this.hash(fs.readFileSync(defaultIcon)),
+				defaultIconRelative = path.relative(this.projectDir, defaultIcon),
+				prevDefaultIcon = previousBuildManifestFiles && previousBuildManifestFiles[defaultIconRelative],
+				prevDefaultIconHash = prevDefaultIcon && prevDefaultIcon.hash,
+
+			this.currentBuildManifestFiles[defaultIconRelative] = {
+				hash: currDefaultIconHash
+			};
+
+			defaultIconChanged = prevDefaultIconHash !== currDefaultIconHash;
+		}
+
+		function copyGenAppIcons() {
```

Comment:
This is sort of confusing. Maybe we just inline this function in the `this.generateAppIcons()` call?